### ISSUE TITLE
Graphql SDL, option to have autogenerated mutations return the type rather than a string

### ISSUE THREAD

**Issue by heresandyboy** [neo4j-graphql/neo4j-graphql#108]
Hi, raising issue from Slack 

We would like the option to specify that any auto generated mutations can return the mutated type as a response rather than a success string as is the only option currently. 

This will give us the option of updating the cache in Apollo client from the response rather than having to perform a refetch query every time. 

We might otherwise rely on a subscription to update the cache - but subscriptions are not implemented yet as far as i'm aware.

_Example mutation from generated schema after posting SDL_
```
type MutationType {
  # Autogenerated
  createAuthor(id: ID, name: String, age: Long): String

 # Would prefer
  createAuthor(id: ID, name: String, age: Long): Author

```

_Slack chat_
> heresandyboy [5:01 PM]
> Is there an easy way to have the auto generated mutations return the type as a response rather than a string with the neo4j-graphql plugin?
> 
> 
> lyonwj [5:09 PM]
> not currently. This was something we wanted to change, but I don't see an issue for it now. Could you create one?
> 
> a current workaround would be to create custom mutations using an `@cypher` directive

**Comment by jorroll** [comment]
It seems like the most future proof solution to this problem is for auto-generated mutations to return a special mutation type. To start, the mutation type could have two fields on it `node` and `result`. The `node` field would return the node that was created or updated, the `result` field would return the same result string that is currently returned. This would necessitate generating custom mutation return types for each mutation.

Or, to start the mutation type could simply have one field on it, `result`, and at a later time additional fields could be added like `node`. This would help push the api to a non-breaking release faster.

This solution would make it easier to accommodate possible future API changes in a non-breaking way, because the mutation type could simply be updated to add new fields.

**Comment by mvid** [comment]
Agreed. Clients like Apollo expect the the modified node returned in a mutation, I believe that is the GraphQL best practice. 

Right now this involves a lot of handrolled mutations to make the backend play nice with GraphQL clients

**Comment by mvid** [comment]
@jexp it’s looking like it will be faster for us to implement this in the plugin instead of hand rolling all of the mutations ourselves. Is there anything to consider if we want to implement this and pass it back upstream?

**Comment by mvid** [comment]
Any update on this? We are looking to get started on it now

**Comment by mvid** [comment]
@jexp any word on this?

**Comment by mvid** [comment]
@heresandyboy I went ahead and implemented this, as my client is currently stuck using this deprecated product: 

https://github.com/mvid/neo4j-graphql/tree/feat/mutation-return

**Comment by jexp** [comment]
Sorry didn’t see your message. 

Did you try the experimental endpoint which does that. 

That will become the default with the next version.
